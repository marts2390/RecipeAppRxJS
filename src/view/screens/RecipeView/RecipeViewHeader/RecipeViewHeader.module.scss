@import "src/view/styles/variables.scss";

.header {
  padding: 1.5em;
  border-radius: 10px;
  background-color: $white;
  margin-bottom: 1em;

  .thumbnailImage {
    background-position: center;
    background-size: cover;
    height: 12em;
    width: 100%;
    border-radius: 10px;
  }

  .headerContent {
    height: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;

    h2 {
      font-size: $font-lg;
      margin-top: 0;
      margin-bottom: 0.25em;
    }

    h3 {
      font-size: $font-md;
      font-weight: $font-medium;
      margin: 0;
    }

    .buttonGroup {
      align-items: center;
      margin-top: 1em;

      @media (min-width: $screen-sm) {
        display: flex;
      }

      @media (min-width: $screen-md) {
        margin-top: 0;
      }

      .homeButton {
        margin-bottom: 1em;

        @media (min-width: $screen-sm) {
          margin-right: 1.5em;
          margin-bottom: 0;
        }
      }

      .button {
        width: 100%;

        @media (min-width: $screen-sm) {
          width: auto;
        }
      }
    }
  }
}
