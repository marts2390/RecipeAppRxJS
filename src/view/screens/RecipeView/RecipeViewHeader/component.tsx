import React, { FC } from 'react'
import { Link } from 'react-router-dom'

import { Grid, Fade, Button } from '@material-ui/core'

import cn from 'classnames'
import styles from './RecipeViewHeader.module.scss'

interface IRecipeViewHeaderComponentProps {
  backgroundImage?: string;
  title?: string;
  category?: string;
  area?: string;
  videoLink?: string;
  isLoaded: boolean;
}

const RecipeViewHeaderComponent: FC<IRecipeViewHeaderComponentProps> = ({
  backgroundImage, title, category, area, videoLink, isLoaded,
}) => (
  <div className={ styles.recipeView }>
    <Fade
      in={ isLoaded }
      timeout={ 500 }
      data-testid="fadeElem"
    >
      <div className={ styles.header }>
        <Grid
          container
          spacing={ 2 }
          alignContent="flex-start"
        >
          <Grid
            item
            xs={ 12 }
            sm={ 4 }
            md={ 3 }
          >
            <div
              className={ styles.thumbnailImage }
              style={ { backgroundImage: `url(${backgroundImage})` } }
            />
          </Grid>
          <Grid
            item
            xs={ 12 }
            sm={ 8 }
            md={ 9 }
          >
            <div className={ styles.headerContent }>
              <div>
                <h2>{title}</h2>
                <h3>{`Area: ${area}`}</h3>
                <h3>{`Category: ${category}`}</h3>
              </div>
              <div className={ styles.buttonGroup }>
                <Link to="/">
                  <Button
                    variant="contained"
                    color="primary"
                    className={ cn(styles.homeButton, styles.button) }
                  >Take me back
                  </Button>
                </Link>
                {videoLink !== '' && (
                <a
                  rel="noreferrer"
                  target="_blank"
                  href={ videoLink }
                >
                  <Button
                    variant="contained"
                    color="secondary"
                    className={ styles.button }
                  >Watch the Method
                  </Button>
                </a>
                )}
              </div>
            </div>
          </Grid>
        </Grid>
      </div>
    </Fade>
  </div>
)

export default RecipeViewHeaderComponent
