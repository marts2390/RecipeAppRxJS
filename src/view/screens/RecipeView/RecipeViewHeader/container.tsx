import React, { FC } from 'react'

import { useObservable } from 'rxjs-hooks'
import UC from 'actions'

import RecipeViewHeaderComponent from './component'


const RecipeViewHeaderContainer:FC = () => {
  const { recipeData } = useObservable(UC.RecipeScreenService.getSubject)
  || UC.RecipeScreenService.getCurrentState()

  return (
    <RecipeViewHeaderComponent
      backgroundImage={ recipeData?.strMealThumb }
      title={ recipeData?.strMeal }
      category={ recipeData?.strCategory }
      area={ recipeData?.strArea }
      videoLink={ recipeData?.strYoutube }
    />
  )
}

export default RecipeViewHeaderContainer
