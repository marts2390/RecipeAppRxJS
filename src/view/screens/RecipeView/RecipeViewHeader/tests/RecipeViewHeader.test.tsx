import React from 'react'
import { shallow } from 'enzyme'
import RecipeViewHeader from '../component'

let wrapper

beforeEach(() => {
  wrapper = shallow(<RecipeViewHeader isLoaded={ false } />)
})

test('Recipe header is rendered', () => {
  expect(wrapper).toMatchSnapshot()
})

test('isLoading is false', () => {
  const fadeElem = wrapper.find({ 'data-testid': 'fadeElem' })
  expect(fadeElem.prop('in')).toBe(false)
})

test('isLoading is true', () => {
  wrapper.setProps({ isLoaded: true })
  const fadeElem = wrapper.find({ 'data-testid': 'fadeElem' })
  expect(fadeElem.prop('in')).toBe(true)
})
