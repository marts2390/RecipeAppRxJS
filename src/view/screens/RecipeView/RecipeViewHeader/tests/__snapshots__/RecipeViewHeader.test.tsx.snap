// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe header is rendered 1`] = `
<div
  className="recipeView"
>
  <ForwardRef(Fade)
    data-testid="fadeElem"
    in={false}
    timeout={500}
  >
    <div
      className="header"
    >
      <WithStyles(ForwardRef(Grid))
        alignContent="flex-start"
        container={true}
        spacing={2}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          md={3}
          sm={4}
          xs={12}
        >
          <div
            className="thumbnailImage"
            style={
              Object {
                "backgroundImage": "url(undefined)",
              }
            }
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          md={9}
          sm={8}
          xs={12}
        >
          <div
            className="headerContent"
          >
            <div>
              <h2 />
              <h3>
                Area: undefined
              </h3>
              <h3>
                Category: undefined
              </h3>
            </div>
            <div
              className="buttonGroup"
            >
              <Link
                to="/"
              >
                <WithStyles(ForwardRef(Button))
                  className="homeButton button"
                  color="primary"
                  variant="contained"
                >
                  Take me back
                </WithStyles(ForwardRef(Button))>
              </Link>
              <a
                rel="noreferrer"
                target="_blank"
              >
                <WithStyles(ForwardRef(Button))
                  className="button"
                  color="secondary"
                  variant="contained"
                >
                  Watch the Method
                </WithStyles(ForwardRef(Button))>
              </a>
            </div>
          </div>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </div>
  </ForwardRef(Fade)>
</div>
`;
